File: 01-basic-operations/node/getUser.js
const AWS = require("aws-sdk");
const DynamoDB = new AWS.DynamoDB({ region: "us-east-1" });

DynamoDB.getItem(
  {
    TableName: "Users",
    Key: { Username: { S: "alexdebrie" } },
  },
  function (err, data) {
    if (err) console.log(`Error fetching user": ${err}`);
    else if (!data.Item) console.log("User does not exist");
    else console.log(`Found user: ${JSON.stringify(data.Item, null, 2)}`);
  }
);

File: 01-basic-operations/python/get_user.py
import json

import boto3

client = boto3.client("dynamodb", region_name="us-east-1")

try:
    response = client.get_item(TableName="Users", Key={"Username": {"S": "alexdebrie"}})
    if response["Item"]:
        print(f"Found user: {json.dumps(response['Item'], indent=2)}")
    else:
        print("User does not exist!")
except Exception as err:
    print(f"Error fetching user: {err}")
